python/s3_bucket_policy_prohibited.py

```diff
@@ -0,0 +1,75 @@
+#
+# This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+#
+# Description: Check if any s3 bucket has bucket policy and if it does mark it non-compliant
+#
+# Trigger Type: Change Triggered
+# Scope of Changes: S3:Instance
+# Accepted Parameters: None
+# Your Lambda function execution role will need to have a policy that provides the appropriate
+# permissions.  Here is a policy that you can consider.  You should validate this for your own
+# environment
+#{
+#    "Version": "2012-10-17",
+#    "Statement": [
+#        {
+#            "Effect": "Allow",
+#            "Action": [
+#                "logs:CreateLogGroup",
+#                "logs:CreateLogStream",
+#                "logs:PutLogEvents"
+#            ],
+#            "Resource": "arn:aws:logs:*:*:*"
+#        },
+#       {
+#            "Effect": "Allow",
+#            "Action": [
+#                "config:PutEvaluations"
+#            ],
+#            "Resource": "*"
+#        }
+#    ]
+#}
+#
+
+import boto3
+import json
+import logging
+
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+
+def evaluate_compliance(configuration_item):
+    bucket_policy = configuration_item["supplementaryConfiguration"].get("BucketPolicy")
```

Comment:
Can you handle cases of 
1) Configuration Items for resource types other than s3 buckets https://github.com/awslabs/aws-config-rules/blob/master/python/s3-exposed-bucket.py#L40
2) Configuration Items for deleted resources https://github.com/awslabs/aws-config-rules/blob/master/python/s3-exposed-bucket.py#L47
Both cases should be not_applicable

Reply:
Added in 214072f